src/app/components/navbar.tsx

```diff
@@ -0,0 +1,121 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import Image from "next/image";
+import { cn } from "@/lib/utils";
+import Button from "./button";
+
+async function getNavBar() {
+	try {
+		const res = await fetch("http://localhost:1337/api/navbars");
+		const { data } = await res.json();
+		return data;
+	} catch (error) {
+		throw new Error("Something went wrong");
+	}
+}
+
+function Navbar() {
+	const [values, setValues] = useState([]);
+	const [hide, setHide] = useState(true);
+
+	useEffect(() => {
+		async function getData() {
+			const response = await getNavBar();
+			setValues(response);
+		}
+		getData();
+	}, []);
+
+	const handleclick = (e: React.MouseEvent<HTMLButtonElement>) => {
+		e.preventDefault();
+		setHide(!hide);
```

Comment:
Esto no es necesario que lo cambies porque no está mal en sí pero para la próxima lo óptimo sería hacer
```suggestion
		setHide(current => !current);
```
Pasando una funcion al setter tenes acceso al valor actual posta posta con toda seguridad.
La otra opción, según el equipo de React puede en algún caso raro tomarte valores desactualizados

> Using the functional update form ensures that you are always working with the latest state.
React might batch state updates, and you can't guarantee that the state is immediately updated. This can lead to unexpected behavior, especially when dealing with multiple state updates in the same function.